### ISSUE TITLE
Module not working for HTTPS-Only HumHub  installations

### ISSUE THREAD

**Issue by daum3ns** [ONLYOFFICE/onlyoffice-humhub#3]
The Module sets the **document.url** incorrect so that a "download failed" alert is seen:
https://stackoverflow.com/questions/46545410/docker-document-server-download-failed

https://api.onlyoffice.com/editors/config/document#url
i see in the logs that the url is set like humhub was plain-http although humhub is setup to use https. there is no way to set this in gui, nor in config files..

when i disable automatic redirect of http traffic to https for humhub, the module works ...

please fix this issue, or let me help you fix it! i think it must be quite simple, but i don't know the framework...

**Comment by Eloncase** [comment]
Hello! Thanks for your contribution.
We are using framework method [Url::to](https://github.com/ONLYOFFICE/onlyoffice-humhub/blob/504a2da95979e54eb54127aba4132c833e86eb68/widgets/EditorWidget.php#L116) to generate links, same method used to generate [permalinks](https://github.com/humhub/humhub/blob/f62bb20fc0fb691e97d43c52cd78d394dbced69f/protected/humhub/modules/content/widgets/PermaLink.php#L35) for example.
It's probably related to [this](https://github.com/humhub/humhub/issues/3815) issue and until it is resolved we can't really tell if this is our problem or not. As soon as issue is resolved we'll make appropriate changes.

**Comment by Eloncase** [comment]
Have you tried making permalink to a post? Is it http or https?

The issue mostly about that `Base URL` setting (`Settings -> General -> Base URL`) should be taken into consideration when generating any kind of URL. Right now it takes current context to do that. 
For example if you set `Base URL` to `https://192.168.1.10` but navigate to humhub using `http://localhost` all links will navigate to `http://localhost` ignoring settings.

Is [Forwarded](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Forwarded) headers configured properly on your proxy server?

**Comment by daum3ns** [comment]
thanks eloncase, i will check this as soon as i have some spare minutes, i think BaseUrl should be correct but the headers i have to check!

**Comment by bha-lockdrives** [comment]
Are there any news? I have the same problems. Humhub is on SSL only.
![onlyoffice](https://user-images.githubusercontent.com/58180257/96414231-20ca9200-11ed-11eb-8279-b20cce0ed6d3.PNG)

**Comment by dreua** [comment]
To add a data point here: My https only installation works fine after some debugging. (Only issue remaining: #28) The installation experience could be improved by more/better error messages or maybe a wizard / some automated tests though.

I have the document server docker image with a port switch: `-p 8080:80` (Earlier mistake: Use port 8080 of the container. Not sure what that is but it surely does not work.) Make sure you can access the document server with your browser via Https. It should display a nicely formatted welcome page ("ONLYOFFICE Docs™ Community Edition – Welcome!").

Apache is setup with virtual hosts for two subdomains, one serving the onlyoffice document server as proxy adding Https ([my configuration](https://gist.github.com/dreua/400faa0be8b0c15484e863b17e62873e)) and the other serving Humhub.

**Comment by daum3ns** [comment]
@bha-lockdrives  when i remember correctly adding this in humhub/protected/config/web.php did the trick for me:
`$_SERVER['HTTPS'] = 'on';`

(see https://github.com/humhub/humhub/issues/3815)

**Comment by smurf1965** [comment]
Hi, recently added DocServer to Humhub. Apparently successfully connected, however when trying to create doument, Humhub throws up 'Could not Onlyoffice Document editor'. Now I think that as Hunhub is on https:// and Digital Ocen Docserver on http://, this could be the issue? Not entirely certain, however any help or suggestion would be greatly appreciated!
![image](https://user-images.githubusercontent.com/73519329/113057074-1b945900-9209-11eb-8823-4bf197234adc.png)
![image](https://user-images.githubusercontent.com/73519329/113057137-2c44cf00-9209-11eb-81d4-5353f648ccf6.png)
![image](https://user-images.githubusercontent.com/73519329/113057187-39fa5480-9209-11eb-80f6-708c6d9cbe91.png)
![image](https://user-images.githubusercontent.com/73519329/113057236-467ead00-9209-11eb-95cf-5f7cc1aa066b.png)

**Comment by dreua** [comment]
> Hunhub is on https:// and Digital Ocen Docserver on http://, this could be the issue?

Yes this is the issue. The other way around works but you shout never attempt to use http resources in a page loaded via https. It creates warnings or does not work at all.

**Comment by smurf1965** [comment]
@dreua Thanks for that, glad to see on the right track! :)

**Comment by dreua** [comment]
@smurf1965 Happy to help! Getting OO+Humhub to work together can really be a p... - well let's call it difficult :wink:
If you have further questions I'd suggest you create a new issue, maybe I can help. In my opinion, appending to already existing issues is only advisable if you are quite sure that you have the same issue *and* you have something valuable to add. (Otherwise do it Youtube-Style: Hit thumbs up on the top post and subscribe! Or create a new issue and link to the existing issue.)

@ someone with moderation privilege: Feel free to hide my comments here.

**Comment by daum3ns** [comment]
argh, i remember having exactly that message.. did you verify the setting i mentioned here: https://github.com/ONLYOFFICE/onlyoffice-humhub/issues/3#issuecomment-767652231

**Comment by daum3ns** [comment]
no i don't have proxy settings set... (although i have a reverse proxy in front, but that's all handled by docker-compose in my case i think, the same reverse proxy is in front of the docserver and translates back to plain http, i.e the docserver itself thinks its plain http when i remember correctly) 

it was a time ago i set this up.. maybe give it a try without autossl... idk..
